node/directives/checkUserAccess.ts

```diff
@@ -3,6 +3,63 @@ import type { GraphQLField } from 'graphql'
 import { defaultFieldResolver } from 'graphql'
 import { SchemaDirectiveVisitor } from 'graphql-tools'
 
+export async function checkUserOrAdminTokenAccess(
+  ctx: Context,
+  operation?: string
+) {
+  const {
+    vtex: { adminUserAuthToken, storeUserAuthToken, logger },
+    clients: { identity, vtexId },
+  } = ctx
+
+  if (!adminUserAuthToken && !storeUserAuthToken) {
+    logger.warn({
+      message: `CheckUserAccess: No admin or store token was provided for ${operation}`,
+      operation,
+    })
+    throw new AuthenticationError('No admin or store token was provided')
+  }
+
+  if (adminUserAuthToken) {
+    try {
+      await identity.validateToken({ token: adminUserAuthToken })
+    } catch (err) {
+      logger.warn({
+        error: err,
+        message: `CheckUserAccess: Invalid admin token for ${operation}`,
+        operation,
+        token: adminUserAuthToken,
```

Comment:
Auth tokens shouldn't be logged, that's sensitive data so it's a security guideline never to log them.

Reply:
Same for the other log messages in this file.

Reply:
I agree, and I will remove it; the problem was in the previous solution.